guide/english/certifications/javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/smallest-common-multiple/index.md

```diff
@@ -156,16 +156,15 @@ Note: If the array only has two elements, then the `for` loop never gets used an
     function smallestCommons(arr) {
 
       // range
-      let min = Math.min.apply(null, arr);
-      let max = Math.max.apply(null, arr);
+      const sorted = arr.sort((a,b) => a > b)
```

Comment:
This `.sort()` method doesnt work. 
```js
[2, 1].sort((a,b) => a > b); //[2, 1]
```
Though you can fix this with changing the callback to something like
```js
[2, 1, 3].sort((a, b) => a > b ? 1 : -1); // [1, 2, 3]
```

Reply:
Hi @ezioda004 thanks for taking a look at this so quickly. I'm not sure where you ran the `sort` method to get that result. But as far as I can tell that `sort` implementation works as intended.

Here it is in a couple different environments:

Repl.it
https://repl.it/@arcdev1/sort-agreaterb

JSBin
https://jsbin.com/zekucob/1/edit?js,console

Reply:
It does not fail.  
```
[2, 1].sort((a,b) => a > b); //[1, 2] not [2, 1]
```

Reply:
@arcdev1 That is very weird, its not being sorted in chrome (70) or even canary (72) but works in firefox.
![screenshot 151](https://user-images.githubusercontent.com/20020273/47120647-13d8d800-d28d-11e8-8573-bf79d22bbc95.png)
And since repl also uses native browser, it doesnt work in that either.
![screenshot 152](https://user-images.githubusercontent.com/20020273/47120690-3ff45900-d28d-11e8-9c93-a3443c519610.png)

Reply:
@arcdev1 Okay, I researched and turns out Chrome 70 is using a new algorithm for sorting (Tim Sort) and they've removed the "bad behavior" of the callback function. 

The callback function needs to return a number or it won't work. What you currently have is a bad comparison function, so I'd suggest changing it to how I've suggested or we'll get conflicting behavior in future.

Reply:
This is a good insight. Thanks.